### ISSUE TITLE
Terminal: emacs shortcuts not working: alt-f, ctrl-f, alt-b, etc.

### ISSUE THREAD

**Issue by bersace** [zed-industries/zed#14543]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Hi,

zed is awesome, thanks.

I'm using bash and used to type `ctrl-f`/`alt-f`, etc. to move inside the readline editor. It's works perfectly in alacritty. However, in Zed, typeing `ctrl-f` inserts `f`. Same for `alt-f`.

I watched this behaviour with the following shortcuts:

- `alt-f`
- `ctrl-f`
- `alt-b`
- `alt-u`

I tried the following keymap without success:

``` json
  {
    "context": "Terminal",
    "bindings": {
      "ctrl-f": ["terminal::SendKeystroke", "ctrl-f"],
      "alt-f": ["terminal::SendKeystroke", "alt-f"]
    }
  }
```

Do you have a clue on what's happening ?

On this subject, default keybings uses a lot `ctrl-k` as a shortcut prefix. This slows down bare `ctrl-k` (kill line) in bash. It seems that the default-linux.json keymaps does not cohabitate well with terminal shortcuts.

Regards,
Étienne

### Environment

```
Zed: v0.143.7 (Zed)
OS: Linux Wayland debian 12
Memory: 15.3 GiB
Architecture: x86_64
```

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>

</pre></details>

**Comment by notpeter** [comment]
Good morning,

Yes, the ctrl-k (cmd-k on mac) is a known outstanding issue:
- #10176

Sorry to hear that `terminal::SendKeystroke` is not functioning for you. Do any ctrl/alt shortcuts work for you in terminal or are they all broken? Can you share some more details about your setup (Desktop Environment, Window Manager, keyboard layout, etc).

You may also want to vote for this issue too:
- https://github.com/zed-industries/zed/issues/14260

**Comment by notpeter** [comment]
I wonder if it might be related to this:
- https://github.com/zed-industries/zed/issues/14053

**Comment by bersace** [comment]
> I wonder if it might be related to this:
> 
> * [Hotkeys do not work if you switch keyboard layouts #14053](https://github.com/zed-industries/zed/issues/14053)

Thanks for the pointer. I don't change keyboard layout while running zed.

**Comment by Tomato6966** [comment]
Experiencing similar issue on windows.:

> I did a cargo build + run on windows.
> 
> When i add a keymap (default is vscode) e.g. "ctrl-k ctrl-f": "editor::Format"
> i first can't format the code when pressing the keymap, because there is the default map "ctrl-k" which focusses and toggles the searchbar, aka i can't create that keymap as i'm used to in vscode forexample
> or when i want to add a comment line command for "ctrl-k ctrl-c" it's still the same aka all keymaps which have a "ctrl-k" in it, don't work, cause the searchbar will always get toggled first
> 
> Trying to override this keymap also doesn't work.

**Comment by notpeter** [comment]
@bersace I merged PR #14556 from @CharlesChen0823 to fix this.

Turns out that change is unnecessary and this can be controlled via [terminal.option_as_meta](https://zed.dev/docs/configuring-zed#option-as-meta) in settings.

In #15535 I'm reverting #14556 and changing the `option_as_meta` default to true.
This will land in Zed Preview later today and in Zed Stable in a week (2024-08-06).

Until then you can add the following to your settings:
```
  "terminal": {
    "option_as_meta": true
  },
```

**Comment by AntonRich** [comment]
@notpeter thanks it worked. I had to delete the key settings though. Turning this option on worked by itself (which surprised me).
